docs/Conditions.md

```diff
@@ -191,4 +211,50 @@ Usage can then look as following:
 #else
 // This renders for desktop platforms
 #endif
-```
\ No newline at end of file
+```
+
+## Conditional Parameters
+
+[Parameter symbols in template](Reference-for-template.json.md#parameter-symbol) can be specified together with optional conditions:
+* [`IsEnabled Condition`](Reference-for-template.json.md#isEnabled) - overwritting presence of input parameter. If condition is specified and evaluates to false (or a `false` constant is passed), passed parameter value (if any) is ignored and processing works as if the parameter value was not passed. This includes application of [default values](Reference-for-template.json.md#default), [verification of mandatory parameters](Reference-for-template.json.md#isRequired), [conditional processing of sources](Conditional-processing-and-comment-syntax.md) and [replacements](Reference-for-template.json.md#replaces).
+* [`IsRequired Condition`](Reference-for-template.json.md#isRequired) - dictates if parameter is mandatory or optional.
+
+### Evaluation
+
+**Input** - currently only other parameter symbols from the template configuration are supported within the parameter conditions. Any other variables are not bound and replaced (they are considered part of literal string).
+
+**Evaluation order** - Dependencies between parameters are detected and evaluation is peformed in order that guarantees that all dependencies are evaluated prior their dependant (see [Topological Sorting](https://en.wikipedia.org/wiki/Topological_sorting) for details).
+
+ In case of cyclic dependency the evaluation proceeds only if current input values of parameters do not lead to indeterministic result (and the cycle is indicated in warning log message). Otherwise an error is reported, indicating the cycle.
+
+ Example `template.json` with cyclic dependency:
+```json
+  "symbols": {
+    "A": {
+      "type": "parameter",
+      "datatype": "bool",
+      "isEnabled": "B != false",
+    },
+    "B": {
+      "type": "parameter",
+      "datatype": "bool",
+      "isEnabled": "A != true",
+    }
+}
+```
+
+Following input parameter values can (and will) be evaluated deterministically: `--A false --B true`
+
+Following input parameter values cannot be evaluated deterministically (and will lead to error): `--A true --B false`
```

Comment:
Not sure if I am following this example:
- `--A false --B true` - leads to A enabled (true != false), B enabled (false != true)
- `--A true --B false` - leads to A disabled (false != false), B disabled (true != true)

Is it somehow related to the case when A and B are not set?

Reply:
Let me add the default values as well - that should make the example less confusing

Reply:
When you do this, can you also show an example of the errors that a user would see in these cases?